src/components/elements/NftSearchBox.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { Box, Button, Input, Flex, Tooltip } from "rimble-ui";
+import { H6 } from "../layouts/LayoutComponents";
+import { Info } from "@rimble/icons";
+import { getNftIdentifiersFromUrl } from "../../utils";
+export interface NftSearchBoxProps {
+  placeholderText?: string;
+  onSearch(value: string, id: string): void;
+}
+
+export const NftSearchBox: React.FunctionComponent<NftSearchBoxProps> = (props) => {
+  const urlParams = new URLSearchParams(window.location.search);
+  const contract = urlParams.get("contract");
+  const tokenId = urlParams.get("tokenId");
+  const [contractAddress, setContractAddress] = useState(contract || "");
+  const [token, setToken] = useState(tokenId || "");
+  const [search, setSearch] = useState("");
+  const [error, setError] = useState("");
+
+  function onKeyDown(event: any) {
```

Comment:
No need to change here but heads up that if you wrap all the inputs in a rimble-ui `<Form>` component you can use its `onSubmit` property to handle these, it'll automatically pick up enter keypress in input fields as well as `<Button type="submit">` clicks and trigger `onSubmit` from any of them.